app/pages/sections/additional-services/order-item/controller.test.js

```diff
@@ -0,0 +1,325 @@
+        itemName: 'item-name',
+        selectedRecipientId: 'fake-recipient-id',
+        serviceRecipientName: 'Some service recipient 1',
+        selectedPriceId: 'some-price-id',
+        selectedPrice,
+      });
+
+      expect(getSelectedPriceManifest.getSelectedPriceManifest.mock.calls.length).toEqual(1);
+      expect(getSelectedPriceManifest.getSelectedPriceManifest).toHaveBeenCalledWith({
+        orderItemType,
+        provisioningType: selectedPrice.provisioningType,
+        type: selectedPrice.type,
+      });
+    });
+
+    it('should call getContext with the correct params', async () => {
+      const selectedPriceManifest = { description: 'fake manifest' };
+      getSelectedPriceManifest.getSelectedPriceManifest.mockReturnValue(selectedPriceManifest);
+
+      await getOrderItemContext({
+        orderId: 'order-1',
+        itemName: 'item-name',
+        odsCode: 'fake-recipient-id',
+        serviceRecipientName: 'Some service recipient 1',
+        selectedPriceId: 'some-price-id',
+        selectedPrice,
+        formData: { price: 0.1 },
+      });
+
+      expect(contextCreator.getContext.mock.calls.length).toEqual(1);
+      expect(contextCreator.getContext).toHaveBeenCalledWith({
+        commonManifest: { title: 'fake manifest' },
+        selectedPriceManifest,
+        odsCode: 'fake-recipient-id',
+        orderId: 'order-1',
+        serviceRecipientName: 'Some service recipient 1',
+        itemName: 'item-name',
+        selectedPrice,
+        formData: { price: 0.1 },
+      });
+    });
+
+    it('should call getContext with the correct params when formData passed in', async () => {
+      const selectedPriceManifest = { description: 'fake manifest' };
+      getSelectedPriceManifest.getSelectedPriceManifest.mockReturnValue(selectedPriceManifest);
+
+      const formData = {
+        quantity: 3,
+        selectEstimationPeriod: 'month',
+        price: 0.1,
+      };
+
+      await getOrderItemContext({
+        orderId: 'order-1',
+        itemName: 'item-name',
+        odsCode: 'fake-recipient-id',
+        serviceRecipientName: 'Some service recipient 1',
+        selectedPriceId: 'some-price-id',
+        selectedPrice,
+        formData,
+      });
+
+      expect(contextCreator.getContext.mock.calls.length).toEqual(1);
+      expect(contextCreator.getContext).toHaveBeenCalledWith({
+        commonManifest: { title: 'fake manifest' },
+        selectedPriceManifest,
+        odsCode: 'fake-recipient-id',
+        orderId: 'order-1',
+        serviceRecipientName: 'Some service recipient 1',
+        itemName: 'item-name',
+        selectedPrice,
+        formData,
+      });
+    });
+  });
+
+  describe('saveSolutionOrderItem', () => {
+    describe('when order item is new', () => {
+      afterEach(() => {
+        postData.mockReset();
+      });
+
+      const serviceRecipient = { name: 'Recipient 1', odsCode: 'ods1' };
+      const solution = { id: 'solutionId1', name: 'Solution 1' };
```

Comment:
For reference solution is also used here. Should it be additionalService?

Reply:
yeah that will be changing in the next PR